test/Core.Test/OrganizationFeatures/OrganizationSubscriptionUpdate/UpdateSecretsManagerSubscriptionCommandTests.cs

```diff
@@ -0,0 +1,631 @@
+        PlanType planType,
+        Guid organizationId,
+        SutProvider<UpdateSecretsManagerSubscriptionCommand> sutProvider)
+    {
+
+        var organization = new Organization
+        {
+            Id = organizationId,
+            SmSeats = 10,
+            SmServiceAccounts = 200,
+            MaxAutoscaleSmSeats = 20,
+            MaxAutoscaleSmServiceAccounts = 300,
+            PlanType = planType,
+            GatewayCustomerId = "1",
+            GatewaySubscriptionId = "2"
+        };
+
+
+
+
+        sutProvider.GetDependency<IOrganizationRepository>()
+            .GetByIdAsync(organizationId)
+            .Returns(organization);
+
+        var organizationUpdate = new OrganizationUpdate
+        {
+            OrganizationId = organization.Id,
+            MaxAutoscaleSeats = 15,
+            SeatAdjustment = 1,
+            MaxAutoscaleServiceAccounts = 300,
+            ServiceAccountsAdjustment = 1
+        };
+
+        sutProvider.GetDependency<IOrganizationRepository>()
+            .GetByIdAsync(organization.Id)
+            .Returns(organization);
+
+        var exception = await Assert.ThrowsAsync<BadRequestException>(() => sutProvider.Sut.UpdateSecretsManagerSubscription(organizationUpdate));
+        Assert.Contains("Existing plan not found", exception.Message, StringComparison.InvariantCultureIgnoreCase);
+    }
+
+    [Theory]
+    [BitAutoData(PlanType.TeamsAnnually)]
+    [BitAutoData(PlanType.TeamsMonthly)]
+    [BitAutoData(PlanType.EnterpriseAnnually)]
+    [BitAutoData(PlanType.EnterpriseMonthly)]
+    public async Task UpdateSecretsManagerSubscription_WithHasAdditionalSeatsOptionfalse_ThrowsBadRequestException(
+        PlanType planType,
+        Guid organizationId,
+        SutProvider<UpdateSecretsManagerSubscriptionCommand> sutProvider)
+    {
+
+        var organization = new Organization
+        {
+            Id = organizationId,
+            SmSeats = 10,
+            SmServiceAccounts = 200,
+            MaxAutoscaleSmSeats = 20,
+            MaxAutoscaleSmServiceAccounts = 300,
+            PlanType = planType,
+            GatewayCustomerId = "1",
+            GatewaySubscriptionId = "2"
+        };
+
+
+
+
+        sutProvider.GetDependency<IOrganizationRepository>()
+            .GetByIdAsync(organizationId)
+            .Returns(organization);
+
+        var organizationUpdate = new OrganizationUpdate
+        {
+            OrganizationId = organization.Id,
+            MaxAutoscaleSeats = 15,
+            SeatAdjustment = 1,
+            MaxAutoscaleServiceAccounts = 300,
+            ServiceAccountsAdjustment = 1
+        };
+
+        sutProvider.GetDependency<IOrganizationRepository>()
+            .GetByIdAsync(organization.Id)
+            .Returns(organization);
+
+        var plan = StaticStore.SecretManagerPlans.Single(x => x.Type == organization.PlanType);
+        plan.HasAdditionalSeatsOption = false;
```

Comment:
Not sure if we should be modifying plan objects like this, even in tests. Either instantiate a new `Plan` object or use the Free plan which should have this already set.

Reply:
The code seems not do recognise the sample plan, I created. any suggestion on plan without using the existing one

Reply:
We discussed injecting `StaticStore` as a dependency, which I think is a good idea, but thinking about it more it's probably a larger change than what we want to make here.

I suggest just setting the `Organization.PlanType` to be a plan where `HasAdditionalSeatsOption == false`, such as the free plan. Then the command will grab that from staticStore and your test should work as intended. You don't actually need to arrange the plan object itself.

Reply:
resolved, i have added the StaticStoreWrapper because I could not use the free plan to test all the cases